template/package.json

```diff
@@ -24,13 +24,13 @@
     "@react-native/eslint-config": "0.77.0-main",
     "@react-native/metro-config": "0.77.0-main",
     "@react-native/typescript-config": "0.77.0-main",
+    "@testing-library/react-native": "^12.8.1",
     "@types/jest": "^29.5.13",
     "@types/react": "^18.2.6",
     "@types/react-test-renderer": "^18.0.0",
     "eslint": "^8.19.0",
     "jest": "^29.6.3",
     "prettier": "2.8.8",
-    "react-test-renderer": "19.0.0-rc-fb9a90fa48-20240614",
```

Comment:
You should not remove `react-test-renderer` as it's required by RNTL. If this works, then only through package manager automatically installing `react-test-renderer` as a peer dep for RNTL. The risk here is that it will pick the wrong version of the package, and `react` and `react-test-renderer` versions need to match.